src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -311,125 +309,81 @@ void FileBasedWal::prepareNewFile(LogID startLogId) {
 }
```

Comment:
maybe we need process the os error more safely and friendly when open file failed, because this error is very common when cpu has a high pressure, crash directly is danger.

Reply:
Agreed.

For simplicity, we used CHECK and LOG(FATAL) too much.

Reply:
We can address this in a separate PR